### ISSUE TITLE
DateTimeFormatter with Locale.ROOT is not working in native image

### ISSUE THREAD

**Issue by udaychandra** [oracle/graal#5510]
**Describe the issue**
A native image built with code that contains datetime formatting using Locale.ROOT throws `java.time.format.DateTimeParseException`.  A super simple example that doesn't work in a native image:

```
  public static void main(String... args) {
    LocalDateTime myDateTime = LocalDateTime.parse("4714-11-24 00:00:00 BC",
      DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss G", Locale.ROOT));
    System.out.println(myDateTime);
  }
```

**Steps to reproduce the issue**
Please include both build steps as well as run steps
1. git clone https://github.com/udaychandra/graalvm-ni-bug-datetime
2. mvn clean package
3. ./target/starter

**Describe GraalVM and your environment:**
 - GraalVM version **GraalVM 22.3.0 Java 17 CE**
 - JDK major version: **17.0.5+8-LTS**
 - OS: **[e.g. macOS Monterey]**
 - Architecture: **x86_64**

**More details**
Here's the stack trace.
```
    Exception in thread "main" java.time.format.DateTimeParseException: Text '4714-11-24 00:00:00 BC' could not be parsed at index 20
	at java.base@17.0.5/java.time.format.DateTimeFormatter.parseResolved0(DateTimeFormatter.java:2052)
	at java.base@17.0.5/java.time.format.DateTimeFormatter.parse(DateTimeFormatter.java:1954)
	at java.base@17.0.5/java.time.LocalDateTime.parse(LocalDateTime.java:494)
```

**Comment by oubidar-Abderrahim** [comment]
Hi, Thank you for reporting this issue, we'll take a look into it shortly

**Comment by oubidar-Abderrahim** [comment]
Tracked internally on GR 42879

**Comment by pejovica** [comment]
Fixed by #6566.
